go/src/socialapi/workers/payment/webhook.go

```diff
@@ -0,0 +1,303 @@
+package payment
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"koding/db/mongodb/modelhelper"
+	"socialapi/workers/email/emailsender"
+
+	"gopkg.in/mgo.v2"
+
+	"github.com/stripe/stripe-go"
+	"github.com/stripe/stripe-go/customer"
+	stripeinvoice "github.com/stripe/stripe-go/invoice"
+)
+
+var mailSender = emailsender.Send
+
+// StripeHandler is the type of handlers for stripe webhook operations
+type StripeHandler func([]byte) error
+
+var stripeActions = map[string]StripeHandler{
+	"charge.succeeded": chargeSucceededHandler,
+	"charge.failed":    chargeFailedHandler,
+
+	"customer.subscription.created":        customerSubscriptionCreatedHandler,
+	"customer.subscription.deleted":        customerSubscriptionDeletedHandler,
+	"customer.subscription.updated":        customerSubscriptionUpdatedHandler,
+	"customer.subscription.trial_will_end": customerSubscriptionTrialWillEndHandler,
+
+	"invoice.created":           invoiceCreatedHandler,
+	"invoice.payment_failed":    invoicePaymentFailedHandler,
+	"invoice.payment_succeeded": invoicePaymentFailedHandler,
+}
+
+// GetHandler returns the registered handler for stripe webhooks if registered
+func GetHandler(name string) (StripeHandler, error) {
+	action, ok := stripeActions[name]
+	if ok {
```

Comment:
when i first wrote it it was returning a default handler, after refactoring, forgot the change the order, fixed it 👍